src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -215,16 +214,29 @@ public virtual void WriteResponseHeaders(OutputFormatterContext context)
             var selectedEncoding = SelectCharacterEncoding(context);
             if (selectedEncoding != null)
             {
-                context.SelectedEncoding = selectedEncoding;
-
                 // Override the content type value even if one already existed.
-                selectedMediaType.Charset = selectedEncoding.WebName;
+                selectedMediaType.Encoding = selectedEncoding;
```

Comment:
Since there's no more `SelectedEncoding` - we just apply it to the CT directly. This is more straightforward than what we used to do :+1: